server/jetstream_api.go

```diff
@@ -131,10 +131,14 @@ const (
 	// jsDirectGetPre
 	jsDirectGetPre = "$JS.API.DIRECT.GET"
 
-	// JSApiConsumerCreate is the endpoint to create ephemeral consumers for streams.
+	// JSApiConsumerCreate is the endpoint to create consumers for streams.
+	// This was also the legacy endpoint for ephemeral consumers.
+	// It now can take consumer name and optional filter subject, which when part of the subject controls access.
 	// Will return JSON response.
-	JSApiConsumerCreate  = "$JS.API.CONSUMER.CREATE.*"
-	JSApiConsumerCreateT = "$JS.API.CONSUMER.CREATE.%s"
+	JSApiConsumerCreate    = "$JS.API.CONSUMER.CREATE.*"
+	JSApiConsumerCreateT   = "$JS.API.CONSUMER.CREATE.%s"
+	JSApiConsumerCreateEx  = "$JS.API.CONSUMER.CREATE.*.>"
```

Comment:
This seems a bit limited/weird, if I wanted to filter on `x.>` then I would need to publish to `$JS.API....CREATE.NAME.x.>` as a subject, something that we do accept, though iirc not in all clients, but it seems very confusing to do this, also not sure how you would write an ACL to match the example destination subject?

I know there's an immediate need to solve this somehow but this doesnt seem like the best way.  

In the short term we really need prioritise the framework to allow arbitrary external policies infront of services so users can write something like OPA policies to achieve this rather than try to weirdly wedge every edge case into this format.  Though sure thats probably longer term than we can wait on this one. I am just weary of putting weird things in here that will not stand the test of time.

Reply:
This should all be hidden by clients, and callouts will be the long term customization effort for sure, but for base level stuff elevating to a subject is our go to.

I am not sure I would want to require external components to be managed just to provide this security.

Reply:
Just to be clear, we are going to allow publishing to subjects containing wildcards?

Reply:
Also durability is now a factor of configuration on the server for the consumer on `inactive_threshold`
An old client on a new server, will not supply the name - so it is durable with the default `inactive_threashold`. Test for durability on clients is effectively checking `inactive_threshold` to be set.

Reply:
No, you can only publish to literals, but in this case we treat wildcards as literal tokens.

Reply:
Old client will supply Durable name which is still supported. If they do not and only use legacy ephemeral API that is also still supported.

Meaning, unless you use the new style nothing changes.

Reply:
so `$JS.API.CONSUMER.CREATE.streamName.consumerName.>` is not how I would create consumerName on streamName with access to all subjects?

Reply:
That is the new style, not the old which again is still supported.

If you want new you can use above, but do not need the trailing ">". Just `$JS.API.CONSUMER.CREATE.streamName.consumerName`

The inactive threshold will determine what we used to define as "durable" vs "ephemeral"